velox/expression/tests/ExpressionFuzzer.cpp

```diff
@@ -590,11 +601,14 @@ ExpressionFuzzer::ExpressionFuzzer(
           continue;
         }
       } else {
+        // TODO Remove this code. argTypes are used only to call
+        // isDeterministic() which can be be figured out without the argTypes.
```

Comment:
@rui-mo Rui, this is a good question. Strictly speaking this is possible, but in practice I don't think it makes sense to define such functions. For a given function name, function registry may have 0 or 1 entry for vector function + 0 or N entries for simple functions. Each of these entries has its own metadata and there is no requirement for all that metadata to match. In practice though it is usually matching. The only exception I'm aware of is IN Presto function. Vector function has default null behavior true, but simple function has default null behavior false.

One option would be to add an API to fetch all registry entries for a given function name. Each entry will be a list of signatures + metadata. If all entries have "deterministic" true, then function is deterministic. Otherwise, not. We can choose to use this API only when argTypes are not yet known.

What do you think?

Reply:
Thanks for the suggestion. I will take a look.
> We can choose to use this API only when argTypes are not yet known

When the argTypes can be generated. Shall we keep using them to determine whether the function if deterministic? Thanks!

Reply:
Perhaps, we can stop generating arg types and check determinism using function name only.

Reply:
Filed https://github.com/facebookincubator/velox/pull/10241 to stop using arg types to check determinism. Thanks.